operator/src/main/java/oracle/kubernetes/operator/Main.java

```diff
@@ -369,14 +370,33 @@ private static Step readExistingPods(String ns) {
   }
 
   static Step readExistingNamespaces(DomainNamespaceSelectionStrategy selectionStrategy,
-                                             Collection<String> domainNamespaces,
-                                             boolean isFullRecheck) {
+                                     Collection<String> domainNamespaces,
+                                     boolean isFullRecheck) {
     CallBuilder builder = new CallBuilder();
     String selector = selectionStrategy.getLabelSelector();
     if (selector != null) {
       builder.withLabelSelectors(selector);
     }
-    return builder.listNamespaceAsync(new NamespaceListStep(selectionStrategy, domainNamespaces, isFullRecheck));
+    return builder.listNamespaceAsync(
```

Comment:
This code is becoming increasingly complex. The coupling between the listNamespace calls and the CRD processing should not be done at this level. One of the great things about the Step/Fiber architecture is that you have those separate steps, and let the code that sets them up define the chain. Putting the dependency directly in the response to the namespace list makes future maintenance ever more difficult.

Reply:
That's fair, but the complexity here really isn't new -- I just moved this code here. I'll see if I can separate them.

Reply:
Okay, I pulled the CRD steps out separately and earlier in the overall flow.